code/game/objects/structures/safe.dm

```diff
@@ -194,7 +257,12 @@ FLOOR SAFES
 
 /obj/structure/safe/attackby(obj/item/I, mob/user, params)
 	if(open)
-		if(I.w_class + space <= maxspace)
+		if(broken && istype(I, /obj/item/safe_internals) && do_after(user, 2 SECONDS, target = src))
+			to_chat(user, "You replace the broken mechanism.")
+			qdel(I)
+			broken = !broken
+			update_icon()

```

Comment:
Probably want to `return` after this block considering we deleted `I`

Reply:
You wouldn't be returning anything but it'll return the proc gracefully if anybody decides to edit this later and add other conditions potentially checking `I` below. Your call.